File: sdk/python/config/ChatResponse.py
############################################################################
#
#  Copyright 2023 Paphus Solutions Inc.
#
#  Licensed under the Eclipse Public License, Version 1.0 (the "License") = None
#  you may not use this file except in compliance with the License.
#  You may obtain a copy of the License at
#
#      http://www.eclipse.org/legal/epl-v10.html
#
#  Unless required by applicable law or agreed to in writing, software
#  distributed under the License is distributed on an "AS IS" BASIS,
#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#  See the License for the specific language governing permissions and
#  limitations under the License.
#
############################################################################
from config.Config import Config
from util.Utils import Writer, Utils


class ChatResponse(Config):
    def __init__(self):
        super().__init__()
        self.message = None
        self.question = None
        self.log = None
        self.conversation = None
        self.emote = None
        self.action = None
        self.pose = None
        self.avatar = None
        self.avatar2 = None
        self.avatar3 = None
        self.avatar4 = None
        self.avatar5 = None
        self.avatarType = None
        self.avatarTalk = None
        self.avatarTalkType = None
        self.avatarAction = None
        self.avatarActionType = None
        self.avatarActionAudio = None
        self.avatarActionAudioType = None
        self.avatarAudio = None
        self.avatarAudioType = None
        self.avatarBackground = None
        self.speech = None
        self.command = None

    def isVideo(self) -> bool:
        return self.avatarType != None and self.avatarType.find("video") != -1

    def isVideoTalk(self):
        return self.avatarTalkType != None and self.avatarTalkType.find("video") != -1

    def parseXML(self, xml):
        root = Utils.loadXML(xml)
        if(root == None):
            return
    
        self.conversation = root.attrib.get("conversation")
        self.emote = root.attrib.get("emote")
        self.action = root.attrib.get("action")
        self.pose = root.attrib.get("pose")
        self.avatar = root.attrib.get("avatar")
        self.avatar2 = root.attrib.get("avatar2")
        self.avatar3 = root.attrib.get("avatar3")
        self.avatar4 = root.attrib.get("avatar4")
        self.avatar5 = root.attrib.get("avatar5")
        self.avatarType = root.attrib.get("avatarType")
        self.avatarTalk = root.attrib.get("avatarTalk")
        self.avatarTalkType = root.attrib.get("avatarTalkType")
        self.avatarAction = root.attrib.get("avatarAction")
        self.avatarActionType = root.attrib.get("avatarActionType")
        self.avatarActionAudio = root.attrib.get("avatarActionAudio")
        self.avatarActionAudioType = root.attrib.get("avatarActionAudioType")
        self.avatarAudio = root.attrib.get("avatarAudio")
        self.avatarAudioType = root.attrib.get("avatarAudioType")
        self.avatarBackground = root.attrib.get("avatarBackground")
        self.speech = root.attrib.get("speech")
        self.command = root.attrib.get("command")
        #tag name
        if(root.find("message") != None):
            self.message = root.find("message").text
        if(root.find("question")):
            self.question = root.find("question").text

File: sdk/nodejs/src/config/ChatResponse.ts
/*
*  Copyright 2023 Paphus Solutions Inc.
*
*  Licensed under the Eclipse Public License, Version 1.0 (the "License");
*  you may not use this file except in compliance with the License.
*  You may obtain a copy of the License at
*
*      http://www.eclipse.org/legal/epl-v10.html
*
*  Unless required by applicable law or agreed to in writing, software
*  distributed under the License is distributed on an "AS IS" BASIS,
*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
*  See the License for the specific language governing permissions and
*  limitations under the License.
*/
import Config from "./Config"
import { XMLReader } from '../util/Utils'

class ChatResponse extends Config {
    message?: string
    question?: string
    log?: string
    conversation?: string
    emote?: string
    action?: string
    pose?: string
    avatar?: string
    avatar2?: string
    avatar3?: string
    avatar4?: string
    avatar5?: string
    avatarType?: string
    avatarTalk?: string
    avatarTalkType?: string
    avatarAction?: string
    avatarActionType?: string
    avatarActionAudio?: string
    avatarActionAudioType?: string
    avatarAudio?: string
    avatarAudioType?: string
    avatarBackground?: string
    speech?: string
    command?: string
    constructor() { super() }

    isVideo(): boolean {
        return this.avatarType != undefined && this.avatarType.includes("video")
    }
    isVideoTalk(): boolean {
        return this.avatarTalkType != undefined && this.avatarTalkType.includes("video")
    }

    parseXML(element: any): void {
        //Attributes
        let reader = new XMLReader(element)
        this.conversation = reader.readAttribute("conversation")
        this.emote = reader.readAttribute("emote")
        this.action = reader.readAttribute("action")
        this.pose = reader.readAttribute("pose")
        this.avatar = reader.readAttribute("avatar")
        this.avatar2 = reader.readAttribute("avatar2")
        this.avatar3 = reader.readAttribute("avatar3")
        this.avatar4 = reader.readAttribute("avatar4")
        this.avatar5 = reader.readAttribute("avatar5")
        this.avatarType = reader.readAttribute("avatarType")
        this.avatarTalk = reader.readAttribute("avatarTalk")
        this.avatarTalkType = reader.readAttribute("avatarTalkType")
        this.avatarAction = reader.readAttribute("avatarAction")
        this.avatarActionType = reader.readAttribute("avatarActionType")
        this.avatarActionAudio = reader.readAttribute("avatarActionAudio")
        this.avatarActionAudioType = reader.readAttribute("avatarActionAudioType")
        this.avatarAudio = reader.readAttribute("avatarAudio")
        this.avatarAudioType = reader.readAttribute("avatarAudioType")
        this.avatarBackground = reader.readAttribute("avatarBackground")
        this.speech = reader.readAttribute("speech")
        this.command = reader.readAttribute("command")
        

        //Elements
        this.message = reader.readElement("message")
        this.question = reader.readElement("question")
    }
}

export default ChatResponse